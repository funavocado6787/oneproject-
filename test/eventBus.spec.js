import { RouterLinkStub, shallowMount, createLocalVue, mount } from '@vue/test-utils'
import EventBus from '../pages/eventBus.vue'
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

Vue.use(Vuetify)
import DisplayCode from '../components/displayCode.vue'
Vue.component(DisplayCode.name, DisplayCode);

const localVue = createLocalVue()
localVue.use(Vuex)

describe('eventBus.vue', () => {
    let store, state;

    beforeEach(() => {
        state = {
            eventBusSnippets: {}
        }
        store = new Vuex.Store({
            modules: {
                codeSnippet: {
                    namespaced: true,
                    actions: {
                        loadEventBusSnippets({commit}) {
                            commit('setEventBusSnippets', {
                                eventBus1: 'eventBus1',
                                eventBus2: 'eventBus2',
                                eventBus3: 'eventBus3'
                             });
                        }
                    },
                    getters: {
                        getEventBusSnippets: state => state.eventBusSnippets
                    },
                    state,
                    mutations: {
                        setEventBusSnippets(state, data) {
                            state.eventBusSnippets = data;
                        }
                    }
                }
            }
        })
    })

    test('Smoke Test - check home page renders by looking for some hard coded text', async () => {
        const wrapper = shallowMount(EventBus, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Event Bus');
    })

    test('eventBus snippet load call was made on creation', async () => {
        const wrapper = shallowMount(EventBus, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.find('h3').text()).toContain('Event Bus');
        expect(Object.keys(state.eventBusSnippets).length).toBe(3);
    })

    test('eventBus made use of snippet 1', async () => {
        const wrapper = mount(EventBus, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('eventBus1');
    })

    test('eventBus made use of snippet 2', async () => {
        const wrapper = mount(EventBus, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('eventBus2');
    })

    test('eventBus made use of snippet 3', async () => {
        const wrapper = mount(EventBus, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        await wrapper.vm.$nextTick()
        expect(wrapper.text()).toContain('eventBus3');
    })


    test('match snapshot', async () => {
        const wrapper = mount(EventBus, {
            stubs: {
                RouterLink: RouterLinkStub
            },
            store, localVue
        })

        expect(wrapper.element).toMatchSnapshot()
    })
})













