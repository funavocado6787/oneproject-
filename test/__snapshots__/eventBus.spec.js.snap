// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eventBus.vue match snapshot 1`] = `
<div
  class="container layout column"
>
  <h3>
    Event Bus
  </h3>
   
  <h4>
    Bus Setup
  </h4>
   
  <p>
    The bus itself is a simple Vue instance with no html.
  </p>
   
  <div
    style="padding-bottom: 1em;"
  >
    <div
      id="app"
    >
      <div
        class="prism-editor-wrapper my-editor"
      >
        <div
          class="prism-editor__container"
        >
          <textarea
            autocapitalize="off"
            autocomplete="off"
            autocorrect="off"
            class="prism-editor__textarea"
            data-gramm="false"
            data-testid="textarea"
            placeholder=""
            spellcheck="false"
          />
          <pre
            class="prism-editor__editor"
            data-testid="preview"
          >
            eventBus1
            <br />
          </pre>
        </div>
      </div>
    </div>
  </div>
   
  <h4>
    Sending an event
  </h4>
   
  <p>
    This is as simple as importing the bus and emiting an event.
  </p>
   
  <div
    id="app"
  >
    <div
      class="prism-editor-wrapper my-editor"
    >
      <div
        class="prism-editor__container"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="prism-editor__textarea"
          data-gramm="false"
          data-testid="textarea"
          placeholder=""
          spellcheck="false"
        />
        <pre
          class="prism-editor__editor"
          data-testid="preview"
        >
          eventBus2
          <br />
        </pre>
      </div>
    </div>
  </div>
   
  <h4>
    Recieving the event
  </h4>
   
  <p>
    On the vue component that needs to recieve the event setup a simple handler.
  </p>
   
  <div
    id="app"
  >
    <div
      class="prism-editor-wrapper my-editor"
    >
      <div
        class="prism-editor__container"
      >
        <textarea
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="prism-editor__textarea"
          data-gramm="false"
          data-testid="textarea"
          placeholder=""
          spellcheck="false"
        />
        <pre
          class="prism-editor__editor"
          data-testid="preview"
        >
          eventBus3
          <br />
        </pre>
      </div>
    </div>
  </div>
</div>
`;
